### PR TITLE
Use CRD Backup and Recover to support failover mechanism

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

* Add the CRD Backup and Recover to support the vineyard cluster's failover mechanism.
* Add relevant e2e test.
* Add the description doc.

Related issue number
--------------------

Related to https://github.com/v6d-io/v6d/issues/1089

### CODE REVIEW

**Comment by dashanji** [comment]
The pr is ready for review, could you please take a look at this? @sighingnow. Once merged, I will autogenerate the helm chart to fix the `autogenerated-helm-chart` test.

File: k8s/main.go [hunk 1]
```diff
@@ -202,8 +229,8 @@ func main() {
 	var schedulerConfigFile string
 	var enableLeaderElection bool
 
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8089", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8090", "The address the probe endpoint binds to.")
 	flag.StringVar(&schedulerConfigFile, "scheduler-config-file", "/etc/kubernetes/scheduler.yaml", "The location of scheduler plugin's configuration file.")
```
Comment (sighingnow) [inline k8s/main.go:234]: Why this change? They should be the default generation.

**Review by sighingnow — APPROVED** [review]
Overall LGTM.

Use `Copyright 2020-2023` for newly created files.

File: k8s/test/e2e/failover-demo/Dockerfile.build-distributed-object-step2 [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+FROM  docker.pkg.github.com/v6d-io/v6d/vineyard-python-dev:latest
+
+WORKDIR /
+
+COPY ./build-distributed-object-step2.py /
+
+ENTRYPOINT [ "python3" ]
+
+CMD [ "build-distributed-object-step2.py" ]
```
Comment (sighingnow) [inline k8s/test/e2e/failover-demo/Dockerfile.build-distributed-object-step2:23]: I'm also thinking about if those jobs can be shared between backup/failover/migration tests to avoid maintaining many similar code pieces.

Comment (sighingnow) [reply k8s/test/e2e/failover-demo/Dockerfile.build-distributed-object-step2:23]: e.g., use the same dockerfile but different args or context directory when building the image.

File: k8s/main.go (hunk 1 above)
Comment (dashanji) [reply k8s/main.go:234]: Oh, it's my fault, Thanks for the catch, I use it for testing locally and forget to change it back.

File: k8s/test/e2e/failover-demo/Dockerfile.build-distributed-object-step2 (hunk 2 above)
Comment (dashanji) [reply k8s/test/e2e/failover-demo/Dockerfile.build-distributed-object-step2]: Done, use one basic Dockerfile to build the different docker images.

**Comment by dashanji** [comment]
I will update the vineyard all copyright in the next pr.
